Feature/lecture list crawling

wjdtjddns98 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #123
- 작업 요약: 인프런 api 가공

## 📄 상세 내용
- [ ] 인프런 open api를 사용해 model의 제약 조건에 맞춰 데이터 가공
- [ ] 모델 제약조건 변경

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항
아직 DB에 저장하는 기능은 미구현 입니다 
## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

SimJunSik commented on apps/lectures/services/inflearn_crawler.py:
```diff
@@ -0,0 +1,156 @@
+                processed_reviews.append(processed_review)
+        return processed_reviews
+
+    except requests.exceptions.RequestException:
+        # 네트워크 오류 발생 시 빈 리스트 반환
+        return []
+    except (json.JSONDecodeError, KeyError):
+        # JSON 파싱 오류나 키 오류 발생 시 빈 리스트 반환
+        return []
```
이 함수랑 아래 함수가 exception 이 발생했을 때 return type 이 다른데,
어쨌든 둘 다 list 형태를 받아오려고 하는 거라 통일하는게 일관성 있을 것 같네요

Umdoong commented on apps/lectures/services/crawling.py:
주석 정리해주세요...ㅠ

Meoyoug commented on apps/lectures/services/inflearn_crawler.py:
프린트문 대신 로거 활용해주세요

Meoyoug reviewed (changes requested):
마이그레이션 파일 손대신 것 같은데 명확한 설명이 없어서 PR 작성란에 기입해주시거나 저한테 설명부탁드립니다..

wjdtjddns98 committed:
✨ feat: inflearn_crawler.py
-inflearn 의 강의 목록 요청와 리뷰 목록 요청 api를 사용해 강의 목록 데이터를 가져오는 model 제약조건에 맞게 Json형식으로 저장하는 기능 구현,
-Json으로 가져올시 강의 link 가 한글을 해쉬해서 가져오는 과정에서 255자를 넘어가는 경우가 발생해  crawled_lectures.py 의 url_link와 thumbnail_img_url 의 max_length 제약조건을 500으로 번경

wjdtjddns98 committed:
✨ feat : migration

- 변경된 crawled_lectures.py 를 바탕으로 migration

wjdtjddns98 committed:
🎨 style: run black run isort, mypy type 어노테이션 추가

wjdtjddns98 committed:
🎨 style: run black

wjdtjddns98 committed:
✨ feat: logging config

wjdtjddns98 committed:
🎨 style : print 대신 logger를 사용

pring 대신 logger를 사용하여 저장

wjdtjddns98 committed:
🎨 style: black, isort 코드 포매터 실행

Meoyoug commented on config/settings/dev.py:
```diff
@@ -34,3 +34,68 @@
+            "class": "logging.StreamHandler",
+            "formatter": "django.server",
+        },
+        "mail_admins": {
```
얘는 운용환경에서만 있으면될거같아서 여기는 지워주세요!

Meoyoug commented on config/settings/dev.py:
```diff
@@ -34,3 +34,68 @@
+    },
+    "loggers": {
+        "django": {
+            "handlers": ["console", "mail_admins"],
```
여기도 메일로 전송되지 않도록 제거해주세요!

Meoyoug replied:
여기 파일핸들러 추가해주세여!

Meoyoug commented on config/settings/dev.py:
```diff
@@ -34,3 +34,68 @@
+            "level": "INFO",
+        },
+        "django.server": {
+            "handlers": ["django.server"],
```
file에 저장되도록 파일핸들러 추가부탁드립니다

Meoyoug commented on apps/lectures/utils/inflearn_crawler.py:
```diff
@@ -0,0 +1,155 @@
+
+import requests
+
+logger = logging.getLogger(__name__)
```
설정에 추가한 로깅을 사용하려면 `__name__` 대신 설정에 추가한 로거 "django" 또는 "django.server"를 가져와야합니다!
if 문 분기로 
```
if not settings.DEBUG:
    logger = logging.getLogger("django")
else:
    logger = logging.getLogger("django.server")
```

이렇게 지정해주세용

wjdtjddns98 committed:
💡 chore: 설정에 추가한 로깅 사용을 위한 수정

wjdtjddns98 committed:
🚚 build: dev.py log관련 email 보내지 않도록 수정

wjdtjddns98 committed:
💡 chore: gitignore에 log 추가

wjdtjddns98 committed:
🐛 fix: Django 환경을 설정하는 코드를 추가하여, 스크립트가 단독으로 실행될 때도 Django의 설정(settings)을 정상적으로 불러올 수있도록 수정

- Django 환경을 설정하는 코드를 추가하여, 스크립트가 단독으로 실행될 때도 Django의 설정(settings)을 정상적으로 불러올 수있도록 수정했습닌다

wjdtjddns98 committed:
🎨 style: black, isort 코드 포매터 실행

다

Meoyoug reviewed (approved):
고생하셨습니다~
